api/src/main/java/org/apache/iceberg/types/Conversions.java

```diff
@@ -72,7 +75,12 @@ public static Object fromPartitionString(Type type, String asString) {
       case DATE:
         return Literal.of(asString).to(Types.DateType.get()).value();
       case TIMESTAMP:
-        return Literal.of(asString).to(Types.TimestampType.withoutZone()).value();
+        if (!asString.contains("T")) {
+          final Instant instant = Timestamp.valueOf(asString).toInstant();
+          return TimeUnit.SECONDS.toMicros(instant.getEpochSecond()) + TimeUnit.NANOSECONDS.toMicros(instant.getNano());
```

Comment:
How do you think we use `asString.replaceFirst(" ", "T")` to make it a ISO standard string that can be parsed?

Reply:
I think both can work, and the current one has been verified, could you share the advantage of doing `asString.replaceFirst(" ", "T")`?

Reply:
It's just simpler and you don't need to have those imports.